Source/Classes/Core/Component/Component.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  Component.swift
+//  Reactant
+//
+//  Created by Tadeáš Kříž on 12/06/16.
+//  Copyright © 2016 Brightify. All rights reserved.
+//
+
+import RxSwift
+
+public protocol Component: class {
+    
+    associatedtype StateType
+    associatedtype ActionType
+    
+    /// DisposeBag for one-time subscriptions made in init. It is reset just before deallocating.
+    var lifetimeDisposeBag: DisposeBag { get }
+    
+    /// DisposeBag for actions in `update`. This is reset before each `update` call.
+    var stateDisposeBag: DisposeBag { get }
+    
+    /// Observable with the current state of the component. Do not use this in `render` to avoid duplicite loading bugs
+    var observableState: Observable<StateType> { get }
+    
+    var previousComponentState: StateType? { get }
+    
+    var componentState: StateType { get set }
```

Comment:
Would adding `var hasComponentState: Bool { get }` make sense to you? I needed something like this earlier and so I thought why not?